src/da/avail.rs

```diff
@@ -4,33 +4,46 @@ use hex::encode;
 use serde_json::json;
 use sp_core::{sr25519, Pair};
 
-use crate::da::da_layers::DALayer;
+use crate::da::da_layers::DaConfig;
 use crate::da::errors::KeyGenError;
 use crate::utils::paths::get_app_home;
 
-pub fn generate_avail_keypair(app_chain: &str) -> Result<DALayer, KeyGenError> {
-    let avail = generate_keypair(app_chain)?;
+pub struct AvailConfig {
+    pub seed: String,
+    pub public_key: String,
+}
+
+impl DaConfig for AvailConfig {
+    fn setup_and_generate_keypair(&self, app_chain: &str) -> Result<(), KeyGenError> {
+        generate_avail_keypair(app_chain)?;
+
+        Ok(())
+    }
+}
+
+fn generate_avail_keypair(app_chain: &str) -> Result<(), KeyGenError> {
+    generate_keypair(app_chain)?;
 
-    Ok(avail)
+    Ok(())
 }
 
-fn generate_keypair(app_chain: &str) -> Result<DALayer, KeyGenError> {
+fn generate_keypair(app_chain: &str) -> Result<(), KeyGenError> {
```

Comment:
all the `generate_keypair` code can be inside `setup_and_generate_keypair`. `generate_avail_keypair` and `generate_keypair` seem unnecessary functions.